### ISSUE TITLE
Feature Request: Named global_step

### ISSUE THREAD

**Issue by sleighsoft** [tensorflow/tensorflow#18218]
I have been developing GAN models for some time now and I find it useful to track the individual stages (pretraining of generator/discriminator, adversarial training) in different global_steps so I can save and restore my actual progress without doing a lot of calculations based on configured training steps and so on.

Therefore, I have basically rebuild the functionality that [tf.train.get_or_create_global_step](https://www.tensorflow.org/versions/master/api_docs/python/tf/train/get_or_create_global_step) provides but with support for named global_steps.

Looks something like `tf.train.get_or_create_global_step(graph, name)`.

Am I the only one finding this particularly helpful?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jart** [comment]
I'm not sure if it would be appropriate to have something like this in TensorFlow core. I'm not sure if we'd have the resources to maintain something like that. There is already a lot of technical debt and complication surrounding the global step, which is a key concept in the many summary logging systems that exist. You might be able to reach out to framework authors like Keras and Estimator to see if they'd be interested. I'm going to close out this issue, but will happily reopen it, provided more community support.
